packages/world/src/systems/MineSystem.sol

```diff
@@ -28,90 +33,160 @@ import { getObjectTypeSchema } from "../utils/ObjectTypeUtils.sol";
 import { TerrainLib } from "./libraries/TerrainLib.sol";
 import { EntityId } from "../EntityId.sol";
 import { PLAYER_MINE_ENERGY_COST } from "../Constants.sol";
+import { ChunkCoord } from "../Types.sol";
+import { COMMIT_EXPIRY_BLOCKS, MAX_COAL, MAX_SILVER, MAX_GOLD, MAX_DIAMOND, MAX_NEPTUNIUM } from "../Constants.sol";
+
+library MineLib {
+  function mineRandomOre(VoxelCoord memory coord) public returns (ObjectTypeId) {
+    ChunkCoord memory chunkCoord = coord.toChunkCoord();
+    uint256 blockNum = OreCommitment._get(chunkCoord.x, chunkCoord.y, chunkCoord.z);
+    require(blockNum > block.number - COMMIT_EXPIRY_BLOCKS, "Ore commitment expired");
```

Comment:
What if some tx fails and the user gets in this state? They cant mine any ores right

Maybe we should add a way to clear commitments?